includes/fbbot/class-omise-fbbot-request-handler.php

```diff
@@ -0,0 +1,157 @@
+	
+	private function __construct() {
+		// Hide the constructor
+	}
+
+	public static function handle_message_from( $sender_id, $user_message ) {
+		$message_store = Omise_FBBot_Message_Store;
+		$message = strtolower( $user_message );
+
+		if ( $message_store::check_greeting_words( $message ) ) {
+			$message = Omise_FBBot_Conversation_Generator::greeting_message( $sender_id );
+			$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
+
+		} else if ( $message_store::check_helping_words( $message ) ) {
+			$helping_message = Omise_FBBot_Conversation_Generator::helping_message();
+			$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $helping_message );
+
+		} else if ( $message_store::check_order_checking( $message ) ) {
+			// Checking order status from order number
+			$checking_text = explode('#', $message);
+
+			if ( ! $checking_text[1] ) {
+			   $rechecking_order_message = Omise_FBBot_Conversation_Generator::rechecking_order_number_message();
+			   $response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $rechecking_order_message );
+				return;
+			}
+
+			$order_id = $checking_text[1];
+			$order_status_message = Omise_FBBot_Conversation_Generator::get_ordet_status_message( $order_id );
+			$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $order_status_message );
+
+		} else {
+			// Handle unrecognize message
+			$unrecognize_message = Omise_FBBot_Conversation_Generator::unrecognized_message();
+			$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $unrecognize_message );
+		}
+	}
+
+	public static function handle_payload_from( $sender_id, $payload ) {
+
+		switch ( $payload ) {
+			case Omise_FBBot_Payload::GET_START_CLICKED:
+				$message = Omise_FBBot_Conversation_Generator::greeting_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
+				break;
+
+			case Omise_FBBot_Payload::FEATURE_PRODUCTS:
+				$message = Omise_FBBot_Conversation_Generator::feature_products_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
+				break;
+
+			case Omise_FBBot_Payload::PRODUCT_CATEGORY:
+				$message = Omise_FBBot_Conversation_Generator::product_category_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
+				break;
+
+			case Omise_FBBot_Payload::CHECK_ORDER:
+				$message = Omise_FBBot_Conversation_Generator::before_checking_order_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
+				break;
+
+			case Omise_FBBot_Payload::HELP:
+				$helping_message = Omise_FBBot_Conversation_Generator::helping_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $helping_message );
+				break;
+
+			default:
+				# Custom payload :
+				self::handle_custom_payload( $sender_id, $payload ); 
+				break;
+		}
```

Comment:
@igroomgrim can move `$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );` out of `switch` condition?

How about
```php
switch ( $payload ) {
	case Omise_FBBot_Payload::GET_START_CLICKED:
		$message = Omise_FBBot_Conversation_Generator::greeting_message( $sender_id );
		break;

	case Omise_FBBot_Payload::FEATURE_PRODUCTS:
		$message = Omise_FBBot_Conversation_Generator::feature_products_message( $sender_id );
		break;

	...

	default:
		# Custom payload :
		$message = self::handle_custom_payload( $sender_id, $payload ); 
		break;
}

$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
```